BuildingBlocks/src/BuildingBlocks.Infrastructure/EventBus/AzureServiceBus/EventBusAzureServiceBus.cs

```diff
@@ -154,7 +163,14 @@ private async Task<bool> ProcessEvent(string eventName, string message)
 
             try
             {
-                await (Task)concreteType.GetMethod("Handle")!.Invoke(handler, new[] { integrationEvent })!;
+                var policy = Policy.Handle<Exception>()
+                .WaitAndRetryAsync(
+                    _pollyRetryCount,
+                    retryAttempt => TimeSpan.FromSeconds(Math.Pow(_minimumBackoff, retryAttempt)),
+                    (ex, _) => _logger.LogWarning(ex.ToString()))
+                .WrapAsync(Policy.TimeoutAsync(_maximumBackoff));
```

Comment:
```c#
retryAttempt => TimeSpan.FromSeconds(Math.Max(Math.Pow(_handlerRetryBehavior.MinimumBackoff, retryAttempt), _handlerRetryBehavior.MaximumBackoff)),
```

I suppose that's what you meant. While that would allow the removal of the Wrap with the Timeout, I'm unsure what advantages it would bring, considering:
1. the current version is more readable, imo.
2. Usage of Wrap/Timeout is readily available.

Or am I missing something else here?

Reply:
1. You can of course expand the `retryAttempt` callback and extract some variables.
2. I'm not even sure if the current behavior is what we want. At least I think that it's not how I interpreted `maximumBackoff`. I think that it defines the maximum waiting time between two calls. And I assume that your solution cancels the whole retry loop if `maximumBackoff` is reached. So for example, if you define:
    minimumBackoff: 2s
    maxiumBackoff: 5s
    retryCount: 3
    it would only do 2 retries, because minimumBackoff^retryCount > maximumBackoff

Btw: I just realized that we have a minor problem. :) What if we set minimumBackoff to 1? This would mean that the timeout is always 1. ;) To avoid this, we should probably convert the minimumBackoff to milliseconds before calculating the power.

Reply:
Ok, I had interpreted `maximumBackoff` as the maximum time we would wait for all retries (regardless of the current retry count). I see now that wasn't the objective.
I'll change things around to fit the expected behaviour (which I have understood now). I'll also address the minimumBackoff 1s problem.